walk-in-interview/src/main/webapp/account/firebase-auth.js

```diff
@@ -0,0 +1,240 @@
+const CurrentLocale = 'en';
+
+firebase.auth().languageCode = CurrentLocale;
+
+// Firebase UI
+const ui = new firebaseui.auth.AuthUI(firebase.auth());
+
+const STRINGS = AppStrings['auth'];
+
+const Auth = {};
+
+/**
+ * This will create a new business account.
+ *
+ * @param {String} email The email for the new business account.
+ * @param {String} password The password for the new business account.
+ */
+Auth.createBusinessAccount = (email, password) => {
+  firebase.auth().createUserWithEmailAndPassword(email, password)
+      .catch((error) => {
+        console.error(error);
+      });
+};
+
+/**
+ * This will sign into an existing business account.
+ *
+ * @param {String} email The email for the exisiting business account.
+ * @param {String} password The password for the existing business account.
+ */
+Auth.signIntoBusinessAccount = (email, password) => {
+  return firebase.auth().signInWithEmailAndPassword(email, password);
+};
+
+/**
+ * This will add the firebase ui for phone authentication
+ * to the provided element.
+ *
+ * @param {String} elementId The div element to add the UI.
+ * @param {String} successPath The url for redirect on login success.
+ * @param {String} newUserInfo The message to be displayed if it is a new user.
+ */
+Auth.addPhoneSignInAndSignUpUI = (elementId, successPath, newUserInfo) => {
+  ui.start(`#${elementId}`, {
+    signInOptions: [
+      {
+        provider: firebase.auth.PhoneAuthProvider.PROVIDER_ID,
+        defaultCountry: 'SG',
+      },
+    ],
+    callbacks: {
+      signInSuccessWithAuthResult: (authResult, redirectUrl) => {
+        // TODO(issue/89): add new user pages for name/skills
+        // for now, this will only redirect to homepage for exisiting users
+
+        // TODO(issue/100): set the cookie at the server side instead
+        setCookie(USER_TYPE_COOKIE_PARAM, USER_TYPE_APPLICANT);
+
+        if (authResult.additionalUserInfo.isNewUser) {
+          // Informs user
+          alert(newUserInfo);
+        }
+
+        return true;
+      },
+
+      signInFailure: (error) => {
+        console.error(error.message);
+
+        alert(STRINGS['sign-in-failure']);
+      }
+    },
+    signInSuccessUrl: successPath,
+  });
+};
+
+/**
+ * Signs out the current user.
+ */
+Auth.signOutCurrentUser = () => {
+  firebase.auth().signOut().then(() => {
+    console.log('sign out successful');
+    // TODO(issue/100): set the cookie at the server side instead
+    setCookie(USER_TYPE_COOKIE_PARAM, USER_TYPE_NO_USER);
+    alert(STRINGS['sign-out-success']);
+  }).catch((error) => {
+    console.error(error);
+  });
+  Auth.subscribeToUserAuthenticationChanges();
+};
+
+/**
+ * Checks the user sign in status.
+ * Makes a POST request to create a session cookie when the user signs in.
+ * Makes a POST request to clear the session cookie when the user signs out.
+ */
+Auth.subscribeToUserAuthenticationChanges = () => {
+  return new Promise((resolve, reject) => {
```

Comment:
i'm not sure i understand this method.

`onAuthStateChanged` from firebase is a subscription: you want to be notified everytime your authentication status changes (login account A > login account B > logout > login C, etc.). This should not `return` anything, it at best it can publish some changes to the rest of the app via a callback.